lib/s3/object_stream.ex

```diff
@@ -0,0 +1,58 @@
+defmodule S3.ObjectStream do
+  @moduledoc """
+  Provide an S3 object as a line-by-line stream.
```

Comment:
I agree, that does seem like a better division of labor and makes testing easier. Do you have any thoughts on how to test the new `BinaryStream` module? Or as I think about it, maybe it would just be testing `ex_aws` library in the end and not really add value?